From 1de5cd43a11e50ad6e74881c53d80aea1f2646da Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 16 Mar 2018 08:16:30 +0100
Subject: [PATCH v2 1/1] efi_loader: use __always_inline for pixel conversion

We optimize for size using -Os so gcc might ignore 'inline'.
Pixel conversions are called so often that we always want to inline them.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
v2
	fix typo in commit message
---
 lib/efi_loader/efi_gop.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/efi_loader/efi_gop.c b/lib/efi_loader/efi_gop.c
index 7b76e49ab01..363ccbb7890 100644
--- a/lib/efi_loader/efi_gop.c
+++ b/lib/efi_loader/efi_gop.c
@@ -56,7 +56,7 @@ static efi_status_t EFIAPI gop_set_mode(struct efi_gop *this, u32 mode_number)
 	return EFI_EXIT(EFI_SUCCESS);
 }
 
-static inline struct efi_gop_pixel efi_vid16_to_blt_col(u16 vid)
+static __always_inline struct efi_gop_pixel efi_vid16_to_blt_col(u16 vid)
 {
 	struct efi_gop_pixel blt = {
 		.reserved = 0,
@@ -70,7 +70,7 @@ static inline struct efi_gop_pixel efi_vid16_to_blt_col(u16 vid)
 	return blt;
 }
 
-static inline u16 efi_blt_col_to_vid16(struct efi_gop_pixel *blt)
+static __always_inline u16 efi_blt_col_to_vid16(struct efi_gop_pixel *blt)
 {
 	return (u16)(blt->red   >> 3) << 11 |
 	       (u16)(blt->green >> 2) <<  5 |
-- 
2.15.1

